script.service.kodi.callbacks/resources/lib/utils/poutil.py

```diff
@@ -304,38 +608,115 @@ def getFileList(self):
         return files_to_scan
 
     def scanPyFilesForStrings(self):
+        """
+        Scans all .py files in directory tree for localized strings
+        :return: List of strings needing localization
+        :rtype: list
+        """
         files = self.getFileList()
         lstrings = []
         for myfile in files:
-            finds = []
-            with open(myfile, 'r') as f:
-                lines = ''.join(f.readlines())
+            with codecs.open(myfile, 'r', 'utf-8') as f:
+                lines = u''.join(f.readlines())
             try:
                 finds = self.find_localizer.findall(lines)
             except re.error:
-                pass
+                finds = []
             finally:
                 if len(finds) != 1:
-                    log(msg='Skipping file: %s, localizer not found' % myfile)
+                    log(msg=u'Skipping file: %s, localizer not found' % myfile)
                 else:
-                    findstr = r"%s\('(.+?)'\)" % finds[0]
+                    findstr = ur"%s\('(.+?)'\)" % finds[0]
                     find = re.compile(findstr)
-                    finds = []
                     try:
                         finds = find.findall(lines)
                     except re.error:
-                        pass
+                        finds = []
                     lstrings += finds
         return lstrings
 
```

Comment:
The whole module has three purposes:
1) Allow me to write gettext _style_ inline string translation using the original English text instead of the kodi numbers (I couldn't figure out what my own code was doing without looking up the number in another file - painful in a largish project).
2) It scans the files in the directory tree looking for other modules where the proper import to use the module is present. If it is, it scans the file for translatable strings and if they aren't in the pofile, they are added.
3) It also allows me to use routines when I am generating my settings.xml file to also add those strings to the po file.

polib looks interesting for help with reading and writing the file, perhaps I'll look into it for the next version, but the current code seems to be working pretty well from what I can tell.